src/tests/blend_tests.cpp

```diff
@@ -22,20 +22,81 @@ static constexpr uint32_t kDefaultDMAColorChannel = 9;
+    {"dstA", NV097_SET_BLEND_FUNC_SFACTOR_V_DST_ALPHA},
+    {"1-dstA", NV097_SET_BLEND_FUNC_SFACTOR_V_ONE_MINUS_DST_ALPHA},
+    {"dstRGB", NV097_SET_BLEND_FUNC_SFACTOR_V_DST_COLOR},
+    {"1-dstRGB", NV097_SET_BLEND_FUNC_SFACTOR_V_ONE_MINUS_DST_COLOR},
+    {"srcAsat", NV097_SET_BLEND_FUNC_SFACTOR_V_SRC_ALPHA_SATURATE},
+    {"cRGB", NV097_SET_BLEND_FUNC_SFACTOR_V_CONSTANT_COLOR},
+    {"1-cRGB", NV097_SET_BLEND_FUNC_SFACTOR_V_ONE_MINUS_CONSTANT_COLOR},
+    {"cA", NV097_SET_BLEND_FUNC_SFACTOR_V_CONSTANT_ALPHA},
+    {"1-cA", NV097_SET_BLEND_FUNC_SFACTOR_V_ONE_MINUS_CONSTANT_ALPHA},
+};
+
+static constexpr struct BlendFunc kBlendFactorsDST[] = {
+    {"0", NV097_SET_BLEND_FUNC_SFACTOR_V_ZERO},
+    {"1", NV097_SET_BLEND_FUNC_SFACTOR_V_ONE},
+    {"srcRGB", NV097_SET_BLEND_FUNC_SFACTOR_V_SRC_COLOR},
+    {"1-srcRGB", NV097_SET_BLEND_FUNC_SFACTOR_V_ONE_MINUS_SRC_COLOR},
+    {"srcA", NV097_SET_BLEND_FUNC_SFACTOR_V_SRC_ALPHA},
+    {"1-srcA", NV097_SET_BLEND_FUNC_SFACTOR_V_ONE_MINUS_SRC_ALPHA},
+    {"dstA", NV097_SET_BLEND_FUNC_SFACTOR_V_DST_ALPHA},
+    {"1-dstA", NV097_SET_BLEND_FUNC_SFACTOR_V_ONE_MINUS_DST_ALPHA},
+    {"dstRGB", NV097_SET_BLEND_FUNC_SFACTOR_V_DST_COLOR},
+    {"1-dstRGB", NV097_SET_BLEND_FUNC_SFACTOR_V_ONE_MINUS_DST_COLOR},
+    {"srcAsat", NV097_SET_BLEND_FUNC_SFACTOR_V_SRC_ALPHA_SATURATE},
+    {"cRGB", NV097_SET_BLEND_FUNC_SFACTOR_V_CONSTANT_COLOR},
+    {"1-cRGB", NV097_SET_BLEND_FUNC_SFACTOR_V_ONE_MINUS_CONSTANT_COLOR},
+    {"cA", NV097_SET_BLEND_FUNC_SFACTOR_V_CONSTANT_ALPHA},
+    {"1-cA", NV097_SET_BLEND_FUNC_SFACTOR_V_ONE_MINUS_CONSTANT_ALPHA},
+};
 
 static constexpr uint32_t kTextureSize = 256;
 static constexpr uint32_t kTexturePitch = kTextureSize * 4;
 
 BlendTests::BlendTests(TestHost &host, std::string output_dir, const Config &config)
     : TestSuite(host, std::move(output_dir), "Blend tests", config) {
-  tests_[kSignedAddName] = [this]() {
-    Test(kSignedAddName, [this]() { Body(NV097_SET_BLEND_EQUATION_V_FUNC_ADD_SIGNED); });
-  };
-  tests_[kSignedRevSubtractName] = [this]() {
-    Test(kSignedRevSubtractName, [this]() { Body(NV097_SET_BLEND_EQUATION_V_FUNC_REVERSE_SUBTRACT_SIGNED); });
-  };
+  for (const auto &test_eqn : kBlendEqns) {
+    for (const auto &test_sfactor : kBlendFactorsSRC) {
+      for (const auto &test_dfactor : kBlendFactorsDST) {
+        uint32_t sfactor = test_sfactor.value;
+        uint32_t dfactor = test_dfactor.value;
+        std::string name = test_sfactor.name;
+        name += "_";
+        name += test_eqn.name;
+        name += "_";
+        name += test_dfactor.name;
+        if (sfactor != NV097_SET_BLEND_FUNC_SFACTOR_V_ZERO ||
+            dfactor != NV097_SET_BLEND_FUNC_SFACTOR_V_ZERO) {
+          tests_[name] = [this, name, &test_eqn, sfactor, dfactor]() {
+            Test(name, [this, name, &test_eqn, sfactor, dfactor]() { Body(test_eqn.value, sfactor, dfactor); });
+          };
+        }    
+      }
+    }
+  }
 }
 
 void BlendTests::Initialize() {
```

Comment:
Now that the tests are more complicated I think it'd be good to also clarify what is being rendered in the message printed to the screen. On line 173, can you put "Alpha: " before the "G: 0x7F" so it's clear that those values are talking about the alpha value for the various colored boxes?